### PR TITLE
[DO NOT merge] Thin opensds

### PR DESCRIPTION
**Pull request by leonwanghui**
**What this PR does / why we need it**:
This PR based on #696 is just to show what's changed in [thin-opensds](https://github.com/opensds/opensds/tree/thin-opensds) branch compared to latest code, please do NOT merge this PR.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: pkg/controller/fileshare/filesharecontroller_test.go [hunk 1]
```diff
@@ -226,3 +226,11 @@ func (fc *fakefileshareClient) AttachVolume(ctx context.Context, in *pb.AttachVo
 func (fc *fakefileshareClient) DetachVolume(ctx context.Context, in *pb.DetachVolumeOpts, opts ...grpc.CallOption) (*pb.GenericResponse, error) {
 	return nil, nil
 }
+
+func (fc *fakefileshareClient) GetMetrics(ctx context.Context, in *pb.GetMetricsOpts, opts ...grpc.CallOption) (*pb.GenericResponse, error) {
+	return nil, nil
+}
+
+func (fc *fakefileshareClient) GetUrls(ctx context.Context, in *pb.NoParams, opts ...grpc.CallOption) (*pb.GenericResponse, error) {
```
Comment (xing-yang) [inline pkg/controller/fileshare/filesharecontroller_test.go]: What are these two added?  They are not specific to Thin OpenSDS.

File: pkg/model/proto/model.proto [hunk 2]
```diff
@@ -147,6 +147,12 @@ service ProvisionDock {
 
     // Collect metrics from metrics driver
     rpc CollectMetrics (CollectMetricsOpts) returns (GenericResponse){}
+
+    // Get metrics from Prometheus
+    rpc GetMetrics (GetMetricsOpts) returns (GenericResponse){}
+
+    // Get 3rd party re-direct URLs for telemetry
+    rpc GetUrls (NoParams) returns (GenericResponse){}
```
Comment (xing-yang) [inline pkg/model/proto/model.proto]: Not related to Thin OpenSDS.

File: pkg/controller/fileshare/filesharecontroller_test.go (hunk 1 above)
Comment (leonwanghui) [reply pkg/controller/fileshare/filesharecontroller_test.go]: I know, but to enable the interface implement function, we have to add these two methods.

Comment (leonwanghui) [reply pkg/controller/fileshare/filesharecontroller_test.go]: Here is an example: https://github.com/opensds/opensds/blob/thin-opensds/pkg/api/controllers/volume.go#L44

Comment (xing-yang) [reply pkg/controller/fileshare/filesharecontroller_test.go]: I see. I think these changes should be submitted to Development branch as a separate bug fix in Capri.

Comment (leonwanghui) [reply pkg/controller/fileshare/filesharecontroller_test.go]: Got it, I will do that

**Comment by kumarashit** [comment]
Closing this PR now. This needs more brainstorming and thoughts

**Pull request closed without merging**